elephant/statistics.py

```diff
@@ -46,8 +46,11 @@ def isi(spiketrain, axis=-1):
     """
     if axis is None:
         axis = -1
-    intervals = np.diff(spiketrain, axis=axis)
-    if hasattr(spiketrain, 'waveforms'):
+    if isinstance(spiketrain, neo.SpikeTrain):
+        intervals = np.diff(
+            np.sort(spiketrain.times.view(pq.Quantity)), axis=axis)
+    else:
+        intervals = np.diff(np.sort(spiketrain), axis=axis)
         intervals = pq.Quantity(intervals.magnitude, units=spiketrain.units)
```

Comment:
This causes an error, since you do not have neo spiketrains anymore but plain numpy arrays.